pkg/cmd/serve.go

```diff
@@ -37,7 +37,7 @@ var (
 	}
 )
 
-func RegisterServeFlags(cmd *cobra.Command, config *server.Config) {
+func MustRegisterServeFlags(cmd *cobra.Command, config *server.Config) {
```

Comment:
I'd suggest avoiding panics in the CLI command:
- `main.go` is handling any error returned by cobra, we can customize error codes returned by the process
- panics do not unwind the stack and do not do any cleaning up, nor goroutines are terminated gracefully
- becomes difficult to log things in the CLI tool and make the output look pretty